include/tvm/Task.h

```diff
@@ -62,8 +62,8 @@ namespace tvm
     TaskDynamicsPtr taskDynamics() const;
     TaskDynamicsPtr secondBoundTaskDynamics() const;  //the dynamics of the upper bound, in the case of double-sided task only.
 
-    template<typename T>
-    std::shared_ptr<typename T::Impl> taskDynamics() const;
+    template<typename T, typename TDImpl = typename T::Impl>
```

Comment:
Not a comment on this particular line, but since we are generalizing this approach, I think the doc on how to implement a task needs to be slightly changed.
Can you either do it or put me a todo on it?

Reply:
I've added some documentation in cba6d21e let me know if this doesn't fit what you had in mind :)